lib/puppet/type/hocon_setting.rb

```diff
@@ -44,7 +44,14 @@
             raise "Type specified as #{@resource[:type]} but was #{value.class}"
           end
         when 'number'
-          unless value.is_a?(Numeric)
+          # Puppet stringifies numerics in versions of Puppet < 4.0.0
+          # Account for this in the type
+          begin
+            numeric_as_string = Integer(value)
```

Comment:
Probably wouldn't harm it - I'll make the change.  Also updating the provider with mk_resource_methods instead of explicit getter/setter, so I'll push those changes and a test this morning.